aws/aws.go

```diff
@@ -253,3 +254,49 @@ func V2ConfigFromURLParams(ctx context.Context, q url.Values) (awsv2.Config, err
 
 	return awsv2cfg.LoadDefaultConfig(ctx, opts...)
 }
+
+// V2s3OptionsFromURLParams returns a slice of s3 options to initialize
+//  an S3 client for AWS SDK v2 initialized based on the URL  parameters in q.
+
+// It is intended to be used by URLOpeners for AWS services if
+// UseV2 returns true.
+//
+// https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/s3#Options
+//
+// It returns an error if q contains any unknown query parameters; callers
+// should remove any query parameters they know about from q before calling
+// V2s3OptionsFromURLParams.
+//
+// The following query options are supported:
+//   - usePathStyle: A value of "true" forces the request to use path-style addressing; sets s3v2.Options.UsePathStyle.
```

Comment:
Just so I understand -- do these options make sense for all AWS services, or just for S3?

If they make sense for all AWS services, let's update the existing V2ConfigFromURLParams to return an awsv2.Config and a []func(*s3v2.Options), instead of creating a new one.

Reply:
`s3v2.Options` options are used by s3 only.

Reply:
So I think we need to have only a separate function.

Reply:
If it's only used by S3, then let's move this code into blob/s3blob. I.e., see how the "accelerate" option is parsed and used.

Reply:
Done that. Please review again.